Introduce `Backend` concept (#615)
alpaka-group/alpaka3: docs/source/basic/terms.rst

Executor
--------

.. _backend:

Backend
-------

> A ``Backend`` is a configuration dictionary (``alpaka::Dict``), typically returned by ``onHost::allBackends(...)``, that provides a ``DeviceSpec`` and an ``Executor``.

Reviewer: ```suggestion
A ``Backend`` the combination of a ``DeviceSpec`` and an ``Executor``. It is typically returned by ``onHost::allBackends(...)``.
```

I would remove some technical details because this part of the documentation is about high level concepts and ideas.
